### ISSUE TITLE
Feature: DH-Key Size

### ISSUE THREAD

**Issue by Der-Orden** [sibiantony/ssleuth#22]
Afer publishing the Logjam-attack[1] DHE looks not good for (P)FS and breakable as easy as RSA as long the same group is used (and 512Bit DH use on over 90% of all server just two primes(!)).
Maybe the rating should be updated but if possible:
Does Firefox gives information about the DH-Keyzise and if yes can ssleuth display it?

Server with broken DH-Keysize to Test 

[1]https://weakdh.org/

**Comment by sibiantony** [comment]
Unfortunately, Firefox doesn't give such info through an interface.

We should raise this in the security channel (Mozilla) or so and see if there is anything planned to expose the DH keysizes through any interface.

**Comment by sibiantony** [comment]
Will keep this open, until we have concrete details on what Mozilla plans to do w.r.t the new attack. If there are more information exposed, I will integrate to ssleuth.

**Comment by tr37ion** [comment]
Did you raise this issue with Mozilla already?

**Comment by sibiantony** [comment]
Nope. If you can, please do and share the bug id. [ Perhaps there is already an existing feature request ]

**Comment by rugk** [comment]
The latest versions of FF now require 1024bit DH. However that's still attackable by a nation state attacker as discussed in https://github.com/sibiantony/ssleuth/issues/38.

Thus I'm also in favour for displaying the DH key size. :+1:

**Comment by sibiantony** [comment]
@rugk I havn't followed this up with Mozilla. Nor have I came across any interfaces that exposes the key size. There are other pressing issues at the moment (e10s) which take too much of my time. If you can take this up with Mozilla and get some status on this 'possible' feature, that'd be great.

If this is made available from any interfaces, many things could be done from SSleuth (make it visible in the panel, optionally block the site etc.) as you suggested.

**Comment by rugk** [comment]
Okay, I've created a bugzilla ticket: https://bugzilla.mozilla.org/show_bug.cgi?id=1215996
Let's see what Mozilla says.
